src/command_buffer.rs

```diff
@@ -218,3 +219,25 @@ struct EntityIndex {
     begin: usize,
     end: usize,
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn populate_archetypes() {
+        let mut world = World::new();
+        let mut buffer = CommandBuffer::new();
+        let ent = world.reserve_entity();
+        let enta = world.reserve_entity();
+        let entb = world.reserve_entity();
+        let entc = world.reserve_entity();
+        buffer.spawn_at(ent, (true, "a"));
+        buffer.spawn_at(entc, (true, "a"));
+        buffer.spawn_at(enta, (1, 1.0));
+        buffer.spawn_at(entb, (1.0, "a"));
+        buffer.run_on(&mut world);
+        let archetypes = world.archetypes_inner().iter();
+        assert_eq!(archetypes.len(), 4);
```

Comment:
Shouldn't `ent` and `entc` land in the same archetype, making this 3 (counting the empty archetype)?

Reply:
before anything spawns  world.archetypes_inner().iter().len() gives 1 after 3 spawns it gives 4, I tested this with normal spawns too which gave the same result

Reply:
You're correct; counting is hard.